### PR TITLE
Refactor/improvements fundamentals

### PR DESCRIPTION
**Pull request by gabisponciano**
What was done: 
1- renaming ai-studio-fundamentals to data-science-and-machine-learning
2- Restructuring of folders 
3- Rewriting of readme files 
4- Docstrings 
5- Adjustments in markdowns 
6-Implementation of MLflow and tensorboard
7- Implementation of Handwritten Digit Classification and Recommender Systems Streamlit

**All the notebooks are ready, Just need to review them again.**
**It is necessary to better analyze the metrics of the notebook a-tale-of-two-cities-analyzing-trends.**

### CODE REVIEW

**Comment by ata-turhan** [comment]
@gabisponciano 

Can you change the names of all 5 projects? The best name for the project is <use_case>_with_<main_tool_used>. So users will know what kind of project they look at exactly.

**Review by ata-turhan — CHANGES_REQUESTED** [review]
New names look really nice.

Now we only need frontends for the related blueprint projects to merge this pr

File: data-science-and-machine-learning/iris_flowers_classification_with_svm/notebooks/iris_flowers_classification_with_svm.ipynb [hunk 1]
```diff
@@ -2468,8 +2492,206 @@
...
+    "model = mlflow.pyfunc.load_model(model_uri=f\"models:/{MODEL_NAME}/{latest_model_version}\")\n",
+    "\n",
+    "df_input = pd.DataFrame({\n",
+    "    'sepal-length': [5.1],\n",
+    "    'sepal-width': [3.5],\n",
+    "    'petal-length':\t[1.4],\n",
+    "    'petal-width': [0.2]\n",
+    "})\n",
+    "prediction = model.predict(df_input)\n",
+    "logger.info(prediction)\n"
```
Comment (NickyJhames) [inline data-science-and-machine-learning/iris_flowers_classification_with_svm/notebooks/iris_flowers_classification_with_svm.ipynb:2694]: As a suggestion of improvement I recommend you to move the Measurement and Accuracy section above the actual cell that prints the model score. 

This will make the notebook more coherent for the reader.

File: data-science-and-machine-learning/iris_flowers_classification_with_svm/README.md
Comment (diogovieir) [inline data-science-and-machine-learning/iris_flowers_classification_with_svm/README.md:1]: Is there any way we can show the image that was inferred to the model? this will add more impact for the reader, will add value.

File: data-science-and-machine-learning/handwritten_digit_classification_with_keras/notebooks/handwritten_digit_classification_with_keras.ipynb
Comment (diogovieir) [inline data-science-and-machine-learning/handwritten_digit_classification_with_keras/notebooks/handwritten_digit_classification_with_keras.ipynb:1]: Is there any way we can show the image that was inferred to the model? this will add more impact for the reader, will add value.

Comment (NickyJhames) [inline data-science-and-machine-learning/handwritten_digit_classification_with_keras/notebooks/handwritten_digit_classification_with_keras.ipynb:1]: Is there any way we can show the image that was inferred to the model? this will add more impact for the reader, will add value.

**Comment by fabrine-d** [comment]
Screenshot 2025-05-19 142051
Screenshot 2025-05-19 161813

File: data-science-and-machine-learning/recommender_systems_with_tensorflow/demo/streamlit-webapp/main.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+    
+# ─────────────────────────────────────────────────────────────
+# 2 ▸ Main – data input
+# ─────────────────────────────────────────────────────────────
+movie_id = st.number_input(
+    "Enter a movie ID:",
+     min_value = 0
+)
+rating = st.number_input(
```
Comment (fabrine-d) [inline data-science-and-machine-learning/recommender_systems_with_tensorflow/demo/streamlit-webapp/main.py:71]: Suggestion: Add appropriate min/max values and help text to guide users. Like: " Enter a rating (1-5): "

File: data-science-and-machine-learning/recommender_systems_with_tensorflow/demo/streamlit-webapp/main.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                "inputs": {"movie_id": [movie_id], "rating":[rating]},
+            }
+            try:
+                response = requests.post(api_url, json=payload, verify=False)
+                response.raise_for_status()
+                data = response.json()
+                
+                # --- Display Results ---
+                if "predictions" in data:
+                    st.success("✅ Here are your vacation recommendations!")
```
[the remark this answers is not in the document]
Comment (gabisponciano) [reply data-science-and-machine-learning/recommender_systems_with_tensorflow/demo/streamlit-webapp/main.py]: done

File: data-science-and-machine-learning/recommender_systems_with_tensorflow/demo/streamlit-webapp/main.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+            box-shadow: 0px 4px 6px rgba(0, 0, 0, 0.1);
+            margin: 10px 0px;
+        }
+    </style>
+""", unsafe_allow_html=True)
+
+# --- Header ---
+st.markdown("<h1 style='text-align: center; color: #2C3E50;'>🎥 Movie Recommendation Agent</h1>", unsafe_allow_html=True)
+
+st.markdown("<h4 style='text-align: center; color: #555;'> Have a movie recommendation based on your movie review.</h3>", unsafe_allow_html=True)
```
[the remark this answers is not in the document]
Comment (gabisponciano) [reply data-science-and-machine-learning/recommender_systems_with_tensorflow/demo/streamlit-webapp/main.py]: done

File: data-science-and-machine-learning/recommender_systems_with_tensorflow/demo/streamlit-webapp/main.py (hunk 2 above)
Comment (gabisponciano) [reply data-science-and-machine-learning/recommender_systems_with_tensorflow/demo/streamlit-webapp/main.py:71]: done for rating

**Review by ata-turhan — APPROVED** [review]
Looks great. Let's merge it 🚀
